kedro/io/core.py

```diff
@@ -254,6 +258,24 @@ def _to_str(obj: Any, is_root: bool = False) -> str:
 
         return f"{type(self).__name__}({_to_str(self._describe(), True)})"
 
+    def _pretty_repr(self, object_description: dict[str, Any]) -> str:
```

Comment:
We discussed using the[ native IPython](https://ipython.readthedocs.io/en/stable/api/generated/IPython.lib.pretty.html) `_repr_pretty_`, but when experiment it's working on my machine but not @ElenaKhaustova .

We decided to move forward with the current implementation.